de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/callout/OrderFastInput.java

```diff
@@ -287,6 +294,22 @@ public static I_C_OrderLine addOrderLine(final Properties ctx, final I_C_Order o
 		}
 		// end: cg: 01717
 
+		final I_C_BPartner partner = ol.getC_BPartner();
+		//
+		final I_C_BPartner_Product bannedProductForPartner = bpProductDAO.getBannedProductForPartner(
+				ol.getM_Product_ID(), 
+				partner.getC_BPartner_ID());
+
+		if (bannedProductForPartner != null)
+		{
+			final String msg = Services.get(IMsgBL.class).getMsg(
+					Env.getCtx(), 
+					BPartnerProductDAO.MSG_ProductSalesBanError, 
```

Comment:
pls move the MSG to interface. Usually the implementation shall be completely hidden.

Reply:
done